src/test/scala/unit/inference/SQLInferenceDataStoreSpec.scala

```diff
@@ -586,6 +586,120 @@ trait SQLInferenceRunnerSpec extends FunSpec with BeforeAndAfter { this: SQLInfe
           .map(rs => rs.long("count")).single.apply().get
         assert(numVariablesInc === 20)
       }
+
+      it("should work with materialization and incremental modes (multinomial)") {
+        cancelUnlessPostgres()
+        // XXX Incremental workflow not supported on Postgres-XL as inference.PostgresInferenceRunner.groundVariables raises the following error:
+        //   org.postgresql.util.PSQLException: ERROR: could not plan this distributed update
+        //   Detail: correlated UPDATE or updating distribution column currently not supported in Postgres-XL.
+        if (dataStoreHelper.isUsingPostgresXL) pending
+        // XXX Incremental workflow not supported on Greenplum as datastore.PostgresDataStore.createTableIfNotExists raising following error:
+        //   org.postgresql.util.PSQLException: ERROR: syntax error at or near "NOT"
+        //   Position: 17
+        if (dataStoreHelper.isUsingGreenplum) pending
+
+        inferenceRunner.init()
+
+        // Insert sample data
+        SQL(s"""CREATE TABLE r1(weight text,
+          is_correct int, id bigint);""").execute.apply()
+        val data = (1 to 100).map { i =>
+          Map("id" -> i, "weight" -> s"weight_${i}", "is_correct" -> i%2)
+        }
+        dataStoreHelper.bulkInsert("r1", data.iterator)
+
+        val dbSettingsMat = DbSettings(dbSettings.driver, dbSettings.url, dbSettings.user,
+          dbSettings.password, dbSettings.dbname, dbSettings.host, dbSettings.port,
+          dbSettings.gphost, dbSettings.gppath, dbSettings.gpport, dbSettings.gpload,
+          IncrementalMode.MATERIALIZATION, null)
+
+        val schema = Map[String, VariableDataType]("r1.is_correct" -> MultinomialType(2))
```

Comment:
Right, what you said can be seen exactly from the test code.  However, the test seems too unrealistic (even the multinomial variable name is `is_correct`!) and makes the reader doubt whether this really is a valid test case.  I was just asking if it could be changed to look more real.  For example, increasing the number from 2 to 3, changing the name to class or category, and fixing the corresponding numbers in the assertions should make it completely look normal.  That's not a huge change, is it?